src/kanata.rs

```diff
@@ -253,20 +310,78 @@ impl Kanata {
         Ok(())
     }
 
-    fn check_handle_layer_change(&mut self) {
+    fn check_handle_layer_change(&mut self, tx: &Sender<EventNotification>) {
         let cur_layer = self.layout.current_layer();
         if cur_layer != self.prev_layer {
+            let new = self.layer_info[cur_layer].name.clone();
             self.prev_layer = cur_layer;
             self.print_layer(cur_layer);
+
+            match tx.try_send(EventNotification::LayerChange { new }) {
+                Ok(_) => {}
+                Err(error) => {
+                    log::error!("could not sent event notification: {}", error);
+                }
+            }
         }
     }
 
     fn print_layer(&self, layer: usize) {
-        log::info!("Entered layer:\n{}", self.layer_strings[layer]);
+        log::info!("Entered layer:\n{}", self.layer_info[layer].cfg_text);
+    }
+
+    pub fn start_notification_loop(kanata: Arc<Mutex<Self>>, rx: Receiver<EventNotification>) {
+        info!("Kanata: entering the event notification loop");
```

Comment:
Need to fix the log message here

Reply:
Changed this to `Kanata: listening for event notifications to relay to connected clients` but I'm open to whatever you wanna put here